node/src/Cardano/Node/API/Swagger.hs

```diff
@@ -0,0 +1,83 @@
+-- necessary for `ToParamSchema Core.EpochIndex`
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Node.API.Swagger where
+
+import           Universum
+
+import           Control.Lens ((?~), at)
+import           Data.Swagger
+import           Servant
+import           Servant.Swagger
+import           Servant.Swagger.UI (SwaggerSchemaUI)
+
+import Pos.Chain.Txp (TxIn, TxOutAux, TxOut)
+import           Pos.Chain.Update (SoftwareVersion)
+import           Pos.Util.Swagger (swaggerSchemaUIServer)
+import           Pos.Web (serveImpl, CConfirmedProposalState)
+import           Pos.Web.Types (TlsParams)
+
+forkDocServer
+    :: HasSwagger a
+    => Proxy a
+    -> SoftwareVersion
+    -> String
+    -> Word16
+    -> Maybe TlsParams
+    -> IO ()
+forkDocServer prxy swVersion ip port' tlsParams =
+    serveImpl
```

Comment:
*serveDocImpl

There's a slight difference between both: we don't check nor require x509 client certificates for the doc server. Otherwise, it will require having setup those certificates in your browser in order to navigate and browse the document. Very unpractical, especially since certificates change from an environment to another.